raster-foundry/raster-foundry
# Add and Use Backsplash Rastersource

## Overview

For a while we've been using a GDAL backed `RasterSource` since it was the only reader that performed well for some particularly large GeoTiffs that we had come across. However, we recently realized that performance of the `GDAL` reader performed poorly when asked to do parallel reads when constructing large mosaics on the fly.

This PR switches to using a custom `RasterSource` that uses cached `GeoTiffInfo` objects to quickly construct a reader for geotiffs.

### Checklist


### Demo

Look at all those parallel reads!
![image](https://user-images.githubusercontent.com/898060/67425287-b6651e80-f5a5-11e9-972d-0dbb4c1a8c3c.png)

### Notes

This is a particularly big changeset, but there wasn't a lot around it -- mostly because we're on 2.x of `GeoTrellis` and all the raster sources are case classes so they can't be extended. Once we switch to 3.0 we can remove this since the new `GeoTiffRasterSource` allows passing in a `GeoTiff` which is what we're doing here basically. Also 3.x includes codecs for `GeoTiffInfo` that were originally from here - so that should be a nice freebie.

Also - overview mosaics were causing some issues and they shouldn't be necessary any longer, so they're removed by this PR. I am going to keep that as its own commit in case we want it back sometime.

## Testing Instructions

- Assemble all jars (including batch), make sure you have a recent DB snapshot, run migrations
- Run the following to backfill metadata (otherwise it will be loaded lazily and you won't get the full awesome affect): `./scripts/console batch "java -cp /opt/raster-foundry/jars/batch-assembly.jar com.rasterfoundry.batch.Main rastersource-metadata-backfill"`
- Start your servers and browse to projects - watch them load, also inspect some traces

Closes #5221
